### ISSUE TITLE
Reproducing paper results: evaluation benchmark, train/test split, and qualitative generalization

### ISSUE THREAD

**Issue by giuli13** [Anytop2025/Anytop#26]
Hi! Thanks for releasing AnyTop and the pretrained models — fantastic work.

I’m running into difficulties reproducing the paper’s numbers and would really appreciate clarification on three points.

## What I did

1. Preprocessed Truebones and downloaded dependencies.
2. Used your pretrained checkpoints to generate 20 samples per character in the benchmark via `sample.generate` (example below).
3. Ran the evaluation code to compute coverage, local/inter diversity, and intra-diversity difference.

Example command I used:

```
python -m sample.generate \
  --model_path save/all_model_dataset_truebones_bs_16_latentdim_128/model000459999.pt \
  --object_type <SKELETON_NAME> \
  --num_repetitions 20 
```

Where we interpret `<SKELETON_NAME>` as each skeleton you want to evaluate for that model, that can be run once per name or loop over the list

Despite following the README, my scores don’t match Table 3 in the paper.

## 1) Exact evaluation procedure to match Table 3

1. Could you please share the precise steps/commands you used to produce the reported metrics?
2. Specific details that would help: Which checkpoint(s) were used for Table 3 — unified all_model or the subset-specific models? 
3. The 30-skeleton benchmark composition: paper says 30 skeletons, randomly selected from those with cumulative frame counts in [600,1200], with distribution 43% Quadrupeds, 17% Bipeds, 23% Flying, 17% Insects. Could you share the exact skeleton list and seed used?
4. Paper notes the number of generated samples per skeleton is proportional to its sample count in the dataset. Do you have the exact per-skeleton sample counts (or a script) used for Table 3? I used a fixed 20/skeleton, which may explain discrepancies. 
5. Metric configuration: temporal window size, stride, nearest-neighbor parameters, and any post-processing used before scoring (e.g., was the foot-contact IK post-process applied to generations when computing metrics, or only for video figures?).

## 2) Train/test split

1. Could you clarify the training split for the unified and subset models?
2. Paper states Truebones Zoo contains ~70 diverse skeletons and that you also train subset-specific models (Bipeds, Quadrupeds, Flying, Millipeds+Snakes). Could you provide the exact training skeleton list for each model, and what (if any) held-out skeletons/motions were excluded from training but included in evaluation? 
3. The README mentions the subset mapping file dataset/truebones/zoo/Truebones_skeletons.txt. Is this the authoritative mapping used for training/eval? 

## 3) Reproducing qualitative results (in-skeleton & cross-skeleton generalization)

1. Which skeletons, model(s), and random seed(s) were used for the in-skeleton composition example (paper mentions chicken pecking/walking)? 
2. For cross-skeleton generalization, could you share the exact pair(s) of source/target skeletons and the command line(s) to reproduce the shown motion and the “nearest ground truth from a different character” visualization? 

Thanks a lot for any pointers — a small script or list (skeletons, seeds, per-skeleton sample counts) would make the evaluation fully reproducible on our side.

**Comment by sigal-raab** [comment]
Dear @giuli13,
We have fixed a bug in our preprocessing code. Please pull the latest changes and rerun the preprocessing script. After the new preprocessing, please rerun the whole evaluation procedure and let us know whether you were able to obtain scores that match the ones in Table 3. 
We are currently very busy with many commitments, so we expect to address your other questions in about two weeks. Our apologies for the delay.
Sigal

**Comment by inbar-2344** [comment]
Hi! Thanks for your interest in our project.
*Evaluation procedure*
1. We’ve refined the instructions in eval/README.md. Please follow them and let us know if anything is unclear.
2. Table 3 reports the metric for the all_model. Results for the subset model appear in the paper’s Appendix.
3. The exact skeletons used to evaluate both the subset and full models are listed in eval/benchmarks.
4. The benchmark skeletons were selected because they each contain 600–1200 frames. For each, we generated 20 samples of length 120 frames.
5. We used the default configuration in the evaluation script.

*Train/test split*
1. Our benchmark is sampled from the training set, following the common practice in works that model the distribution of unique data (e.g., SinMDM). For instance, in-skeleton generalization is evaluated specifically on skeletons that were included during training.
2. All skeletons used for training are listed in dataset/truebones/zoo/Truebones_skeletons.txt. The skeletons used for evaluation are the ones referenced in Evaluation procedure item 3 (eval/benchmarks).
3. Yes—the split follows the specification above.

*Reproducing qualitative results (in-skeleton & cross-skeleton generalization)*
As Sigal noted, there was a bug in the pre-processing script that affected output quality (including the generated cond.npy). Please pull the latest code and regenerate samples. If anything is still unclear after that, feel free to reach out.

**Comment by xiaozhi-123** [comment]
Could you possibly give me a copy of the Truebones FBX/.BVH ZOO? I would be extremely grateful.

It's just too expensive. The price of $195 USD (which is about 1,386.54 RMB) is something I really can't afford as a student, and our lab is very poor too. My Gmail address is daiduc349@gmail.com.

Thank you so much.
